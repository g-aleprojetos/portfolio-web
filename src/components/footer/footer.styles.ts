import styled from 'styled-components';
import {Image} from 'components/image';
import {margin} from 'resources/margins';
import {media} from 'resources/media';
import {Text} from 'components/text';
import {colors} from 'resources/colors';

type PropsExtra = {
  $isActivated?: boolean;
  $backgroundblack?: boolean;
  $position?: number;
};

export const Container = styled.div<PropsExtra>`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 100%;
  border-top-left-radius: ${margin.xsmall}px;
  border-top-right-radius: ${margin.xsmall}px;
  background-color: ${({$backgroundblack}) =>
    $backgroundblack ? `${colors.darkgray}` : `${colors.background01}`};
  z-index: 10;
`;

export const ContainerFooter = styled.footer`
  display: none;
  position: fixed;
  bottom: 0;
  width: 100%;
  height: 70px;
  border-top: 10px solid ${colors.darkgray};
  background: ${colors.darkgray};

  @media screen and (${media.mobile}) {
    display: flex;
  }
`;

export const ContainerItem = styled.ul`
  display: flex;
  position: relative;
  width: 350px;
`;

export const Imagem = styled(Image).attrs<PropsExtra>(
  ({$isActivated, $backgroundblack}) => ({
    largura: margin.large,
    altura: margin.large,
    color: $isActivated
      ? colors.darkgray
      : $backgroundblack
        ? colors.background01
        : colors.darkgray,
  }),
)`
  transition: 0.3s;
`;

export const Item = styled.li<PropsExtra>`
  position: relative;
  list-style: none;
  width: 90px;
  height: 70px;
  z-index: 1;
  transform: ${({$isActivated}) =>
    $isActivated ? 'translateY(-40px)' : 'translateY(0px)'};
  transition: 0.5s;
`;

export const Navegacao = styled.a`
  display: flex;
  position: relative;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  width: 100%;
  height: 100%;
  text-decoration: none;
`;

export const TextNavegation = styled(Text).attrs<PropsExtra>(
  ({$backgroundblack}) => ({
    color: $backgroundblack ? colors.background01 : colors.darkgray,
  }),
)`
  display: flex;
  position: absolute;
  justify-content: center;
  align-items: center;
  width: 100%;
  font-weight: 400;
  letter-spacing: 0.05em;
  transition: 0.5s;
  opacity: ${({$isActivated}) => ($isActivated ? 1 : 0)};
  transform: ${({$isActivated}) =>
    $isActivated ? 'translateY(48px)' : 'translateY(20px)'};
`;

export const IndicatorPosition = (indicador?: number) => {
  switch (indicador) {
    case 0:
      return 'translateX(1px)';
    case 1:
      return 'translateX(88px)';
    case 2:
      return 'translateX(176px)';
    case 3:
      return 'translateX(264px)';
  }
};

export const Indicator = styled.div<PropsExtra>`
  display: ${({$position}) => ($position === -1 ? 'none' : 'block')};
  position: absolute;
  top: -60%;
  left: 2%;
  width: 70px;
  height: 70px;
  border-radius: 50%;
  background-color: ${colors.vert};
  border: 6px solid ${colors.darkgray};
  box-sizing: border-box;
  transition: transform 0.5s;
  transform: ${({$position}) => IndicatorPosition($position)};

  &::before {
    content: '';
    position: absolute;
    top: 62%;
    left: -18px;
    width: 18px;
    height: 18px;
    background: transparent;
    border-top-right-radius: 20px;
    box-shadow: 0px -10px 0 0 ${colors.darkgray};
  }

  &::after {
    content: '';
    position: absolute;
    top: 62%;
    right: -18px;
    width: 18px;
    height: 18px;
    background: transparent;
    border-top-left-radius: 20px;
    box-shadow: 0px -10px 0 0 ${colors.darkgray};
  }
`;
