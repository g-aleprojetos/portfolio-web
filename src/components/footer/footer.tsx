import React from 'react';
import IconHome from 'assets/icons/iHome.svg';
import IconPerson from 'assets/icons/iUser.svg';
import IconDashBoard from 'assets/icons/iDashBoard.svg';
import IconSettings from 'assets/icons/iSettings.svg';
import routes from 'resources/routes';
import {useTranslation} from 'react-i18next';
import {namespaces} from 'utils/i18n/i18n.constants';
import 'utils/i18n';
import {useBackgroundContext} from 'context/background';
import {useLocation} from 'react-router-dom';
import * as S from './footer.styles';

export const Footer = () => {
  const {themeDark} = useBackgroundContext();
  const {t} = useTranslation(namespaces.pages.header);
  const location = useLocation();

  return (
    <S.ContainerFooter data-testid="footer">
      <S.Container data-testid="container-footer" $backgroundblack={themeDark}>
        <S.ContainerItem>
          <S.Item $isActivated={location.pathname === routes.Home}>
            <S.Navegacao to={routes.Home}>
              <S.Imagem
                data-testid="test-image-icon-home"
                alt="Home"
                src={IconHome}
                $isActivated={location.pathname === routes.Home}
                $backgroundblack={themeDark}
              />
              <S.TextNavegation
                $isActivated={location.pathname === routes.Home}
                $backgroundblack={themeDark}>
                {t('home')}
              </S.TextNavegation>
            </S.Navegacao>
          </S.Item>
          <S.Item $isActivated={location.pathname === routes.Projects}>
            <S.Navegacao to={routes.Projects}>
              <S.Imagem
                data-testid="test-image-icon-projects"
                alt="Projects"
                src={IconDashBoard}
                $isActivated={location.pathname === routes.Projects}
                $backgroundblack={themeDark}
              />
              <S.TextNavegation
                $isActivated={location.pathname === routes.Projects}
                $backgroundblack={themeDark}>
                {t('projects')}
              </S.TextNavegation>
            </S.Navegacao>
          </S.Item>
          <S.Item $isActivated={location.pathname === routes.About}>
            <S.Navegacao to={routes.About}>
              <S.Imagem
                data-testid="test-image-icon-about"
                alt="About"
                src={IconPerson}
                $isActivated={location.pathname === routes.About}
                $backgroundblack={themeDark}
              />
              <S.TextNavegation
                $isActivated={location.pathname === routes.About}
                $backgroundblack={themeDark}>
                {t('about')}
              </S.TextNavegation>
            </S.Navegacao>
          </S.Item>
          <S.Item $isActivated={location.pathname === routes.Settings}>
            <S.Navegacao to={routes.Settings}>
              <S.Imagem
                data-testid="test-image-icon-settings"
                alt="Settings"
                src={IconSettings}
                $isActivated={location.pathname === routes.Settings}
                $backgroundblack={themeDark}
              />
              <S.TextNavegation
                $isActivated={location.pathname === routes.Settings}
                $backgroundblack={themeDark}>
                {t('settings')}
              </S.TextNavegation>
            </S.Navegacao>
          </S.Item>
          <S.Indicator $position={getRouteIndex(location.pathname)} />
        </S.ContainerItem>
      </S.Container>
    </S.ContainerFooter>
  );
};

const getRouteIndex = (pathname: string): number => {
  const routeKeys = Object.keys(routes) as Array<keyof typeof routes>;
  const index = routeKeys.findIndex(key => routes[key] === pathname);

  return index;
};
