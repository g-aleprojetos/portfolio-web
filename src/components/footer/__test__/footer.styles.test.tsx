import React from 'react';
import {render} from '@testing-library/react';
import {MemoryRouter} from 'react-router-dom';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import * as S from '../footer.styles';

describe('Footer.styles', () => {
  test('Container DEVE ser igual ao snapshot QUANDO $backgroundblack for igual a true', () => {
    const {container} = render(<S.Container $backgroundblack={true} />);
    expect(container).toMatchSnapshot();
  });

  test('Container DEVE ser igual ao snapshot QUANDO $backgroundblack for igual a false', () => {
    const {container} = render(<S.Container $backgroundblack={false} />);
    expect(container).toMatchSnapshot();
  });

  test('ContainerItem DEVE ser igual ao snapshot', () => {
    const {container} = render(<S.ContainerItem />);
    expect(container).toMatchSnapshot();
  });

  test('Imagem DEVE ser igual ao snapshot com $isActivated true', () => {
    const {container} = render(<S.Imagem src="icon.svg" $isActivated={true} />);
    expect(container).toMatchSnapshot();
  });

  test('Imagem DEVE ser igual ao snapshot com $backgroundblack true', () => {
    const {container} = render(
      <S.Imagem src="icon.svg" $backgroundblack={true} />,
    );
    expect(container).toMatchSnapshot();
  });

  test('Item DEVE ser igual ao snapshot com $isActivated true', () => {
    const {container} = render(<S.Item $isActivated={true} />);
    expect(container).toMatchSnapshot();
  });

  test('Navegacao DEVE ser igual ao snapshot', () => {
    const {container} = render(
      <MemoryRouter>
        <S.Navegacao to="/" />
      </MemoryRouter>,
    );
    expect(container).toMatchSnapshot();
  });

  test('TextNavegation DEVE ser igual ao snapshot com $isActivated true', () => {
    const {container} = render(<S.TextNavegation $isActivated={true} />);
    expect(container).toMatchSnapshot();
  });

  test('TextNavegation DEVE ser igual ao snapshot QUANDO $backgroundblack for igual a true', () => {
    const {container} = render(<S.TextNavegation $backgroundblack={true} />);
    expect(container).toMatchSnapshot();
  });

  test('TextNavegation DEVE ser igual ao snapshot QUANDO $backgroundblack for igual a false', () => {
    const {container} = render(<S.TextNavegation $backgroundblack={false} />);
    expect(container).toMatchSnapshot();
  });

  test('Indicator DEVE ser igual ao snapshot com $position 0', () => {
    const {container} = render(<S.Indicator $position={0} />);
    expect(container).toMatchSnapshot();
  });

  test('Indicator DEVE ser igual ao snapshot com $position 1', () => {
    const {container} = render(<S.Indicator $position={1} />);
    expect(container).toMatchSnapshot();
  });

  test('Indicator DEVE ser igual ao snapshot com $position 2', () => {
    const {container} = render(<S.Indicator $position={2} />);
    expect(container).toMatchSnapshot();
  });

  test('Indicator DEVE ser igual ao snapshot com $position 3', () => {
    const {container} = render(<S.Indicator $position={3} />);
    expect(container).toMatchSnapshot();
  });

  test('Indicator DEVE ser igual ao snapshot com $position -1', () => {
    const {container} = render(<S.Indicator $position={-1} />);
    expect(container).toMatchSnapshot();
  });
});
