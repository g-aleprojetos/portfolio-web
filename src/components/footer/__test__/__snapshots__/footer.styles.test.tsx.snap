// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer.styles Container DEVE ser igual ao snapshot QUANDO $backgroundblack for igual a false 1`] = `
.c0 {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 100%;
  border-top-left-radius: 8px;
  border-top-right-radius: 8px;
  background-color: #ffffff;
  z-index: 10;
}

<div>
  <div
    class="c0"
  />
</div>
`;

exports[`Footer.styles Container DEVE ser igual ao snapshot QUANDO $backgroundblack for igual a true 1`] = `
.c0 {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 100%;
  border-top-left-radius: 8px;
  border-top-right-radius: 8px;
  background-color: #23242a;
  z-index: 10;
}

<div>
  <div
    class="c0"
  />
</div>
`;

exports[`Footer.styles ContainerItem DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: flex;
  position: relative;
  width: 350px;
}

<div>
  <ul
    class="c0"
  />
</div>
`;

exports[`Footer.styles Imagem DEVE ser igual ao snapshot com $backgroundblack true 1`] = `
.c0 {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 32px;
  height: 32px;
  box-sizing: border-box;
}

.c2 {
  display: flex;
  justify-content: center;
  align-items: center;
  pointer-events: auto;
  cursor: default;
}

.c1 {
  transition: 0.3s;
}

<div>
  <div
    class="c0 c1"
    data-testid="container-image"
  >
    <svg
      class="c2"
      color="#ffffff"
      data-testid="svg"
      src="icon.svg"
      style="fill: #ffffff; transform: rotate(0deg);"
    />
  </div>
</div>
`;

exports[`Footer.styles Imagem DEVE ser igual ao snapshot com $isActivated true 1`] = `
.c0 {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 32px;
  height: 32px;
  box-sizing: border-box;
}

.c2 {
  display: flex;
  justify-content: center;
  align-items: center;
  pointer-events: auto;
  cursor: default;
}

.c1 {
  transition: 0.3s;
}

<div>
  <div
    class="c0 c1"
    data-testid="container-image"
  >
    <svg
      class="c2"
      color="#23242a"
      data-testid="svg"
      src="icon.svg"
      style="fill: #23242a; transform: rotate(0deg);"
    />
  </div>
</div>
`;

exports[`Footer.styles Indicator DEVE ser igual ao snapshot com $position -1 1`] = `
.c0 {
  display: none;
  position: absolute;
  top: -60%;
  left: 2%;
  width: 70px;
  height: 70px;
  border-radius: 50%;
  background-color: #29fd53;
  border: 6px solid #23242a;
  box-sizing: border-box;
  transition: transform 0.5s;
}

.c0::before {
  content: '';
  position: absolute;
  top: 62%;
  left: -18px;
  width: 18px;
  height: 18px;
  background: transparent;
  border-top-right-radius: 20px;
  box-shadow: 0px -10px 0 0 #23242a;
}

.c0::after {
  content: '';
  position: absolute;
  top: 62%;
  right: -18px;
  width: 18px;
  height: 18px;
  background: transparent;
  border-top-left-radius: 20px;
  box-shadow: 0px -10px 0 0 #23242a;
}

<div>
  <div
    class="c0"
  />
</div>
`;

exports[`Footer.styles Indicator DEVE ser igual ao snapshot com $position 0 1`] = `
.c0 {
  display: block;
  position: absolute;
  top: -60%;
  left: 2%;
  width: 70px;
  height: 70px;
  border-radius: 50%;
  background-color: #29fd53;
  border: 6px solid #23242a;
  box-sizing: border-box;
  transition: transform 0.5s;
  transform: translateX(1px);
}

.c0::before {
  content: '';
  position: absolute;
  top: 62%;
  left: -18px;
  width: 18px;
  height: 18px;
  background: transparent;
  border-top-right-radius: 20px;
  box-shadow: 0px -10px 0 0 #23242a;
}

.c0::after {
  content: '';
  position: absolute;
  top: 62%;
  right: -18px;
  width: 18px;
  height: 18px;
  background: transparent;
  border-top-left-radius: 20px;
  box-shadow: 0px -10px 0 0 #23242a;
}

<div>
  <div
    class="c0"
  />
</div>
`;

exports[`Footer.styles Indicator DEVE ser igual ao snapshot com $position 1 1`] = `
.c0 {
  display: block;
  position: absolute;
  top: -60%;
  left: 2%;
  width: 70px;
  height: 70px;
  border-radius: 50%;
  background-color: #29fd53;
  border: 6px solid #23242a;
  box-sizing: border-box;
  transition: transform 0.5s;
  transform: translateX(88px);
}

.c0::before {
  content: '';
  position: absolute;
  top: 62%;
  left: -18px;
  width: 18px;
  height: 18px;
  background: transparent;
  border-top-right-radius: 20px;
  box-shadow: 0px -10px 0 0 #23242a;
}

.c0::after {
  content: '';
  position: absolute;
  top: 62%;
  right: -18px;
  width: 18px;
  height: 18px;
  background: transparent;
  border-top-left-radius: 20px;
  box-shadow: 0px -10px 0 0 #23242a;
}

<div>
  <div
    class="c0"
  />
</div>
`;

exports[`Footer.styles Indicator DEVE ser igual ao snapshot com $position 2 1`] = `
.c0 {
  display: block;
  position: absolute;
  top: -60%;
  left: 2%;
  width: 70px;
  height: 70px;
  border-radius: 50%;
  background-color: #29fd53;
  border: 6px solid #23242a;
  box-sizing: border-box;
  transition: transform 0.5s;
  transform: translateX(176px);
}

.c0::before {
  content: '';
  position: absolute;
  top: 62%;
  left: -18px;
  width: 18px;
  height: 18px;
  background: transparent;
  border-top-right-radius: 20px;
  box-shadow: 0px -10px 0 0 #23242a;
}

.c0::after {
  content: '';
  position: absolute;
  top: 62%;
  right: -18px;
  width: 18px;
  height: 18px;
  background: transparent;
  border-top-left-radius: 20px;
  box-shadow: 0px -10px 0 0 #23242a;
}

<div>
  <div
    class="c0"
  />
</div>
`;

exports[`Footer.styles Indicator DEVE ser igual ao snapshot com $position 3 1`] = `
.c0 {
  display: block;
  position: absolute;
  top: -60%;
  left: 2%;
  width: 70px;
  height: 70px;
  border-radius: 50%;
  background-color: #29fd53;
  border: 6px solid #23242a;
  box-sizing: border-box;
  transition: transform 0.5s;
  transform: translateX(264px);
}

.c0::before {
  content: '';
  position: absolute;
  top: 62%;
  left: -18px;
  width: 18px;
  height: 18px;
  background: transparent;
  border-top-right-radius: 20px;
  box-shadow: 0px -10px 0 0 #23242a;
}

.c0::after {
  content: '';
  position: absolute;
  top: 62%;
  right: -18px;
  width: 18px;
  height: 18px;
  background: transparent;
  border-top-left-radius: 20px;
  box-shadow: 0px -10px 0 0 #23242a;
}

<div>
  <div
    class="c0"
  />
</div>
`;

exports[`Footer.styles Item DEVE ser igual ao snapshot com $isActivated true 1`] = `
.c0 {
  position: relative;
  list-style: none;
  width: 90px;
  height: 70px;
  z-index: 1;
  transform: translateY(-40px);
  transition: 0.5s;
}

<div>
  <li
    class="c0"
  />
</div>
`;

exports[`Footer.styles Navegacao DEVE ser igual ao snapshot 1`] = `
.c0 {
  display: flex;
  position: relative;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  width: 100%;
  height: 100%;
  text-decoration: none;
}

<div>
  <a
    class="c0"
  />
</div>
`;

exports[`Footer.styles TextNavegation DEVE ser igual ao snapshot QUANDO $backgroundblack for igual a false 1`] = `
.c0 {
  font-size: 12px;
  font-family: AvertaStd-Regular;
  color: #23242a;
  margin-top: 0px;
  text-transform: none;
  cursor: text;
}

.c1 {
  display: flex;
  position: absolute;
  justify-content: center;
  align-items: center;
  width: 100%;
  font-weight: 400;
  letter-spacing: 0.05em;
  transition: 0.5s;
  opacity: 0;
  transform: translateY(20px);
}

<div>
  <p
    class="c0 c1"
    color="#23242a"
  />
</div>
`;

exports[`Footer.styles TextNavegation DEVE ser igual ao snapshot QUANDO $backgroundblack for igual a true 1`] = `
.c0 {
  font-size: 12px;
  font-family: AvertaStd-Regular;
  color: #ffffff;
  margin-top: 0px;
  text-transform: none;
  cursor: text;
}

.c1 {
  display: flex;
  position: absolute;
  justify-content: center;
  align-items: center;
  width: 100%;
  font-weight: 400;
  letter-spacing: 0.05em;
  transition: 0.5s;
  opacity: 0;
  transform: translateY(20px);
}

<div>
  <p
    class="c0 c1"
    color="#ffffff"
  />
</div>
`;

exports[`Footer.styles TextNavegation DEVE ser igual ao snapshot com $isActivated true 1`] = `
.c0 {
  font-size: 12px;
  font-family: AvertaStd-Regular;
  color: #23242a;
  margin-top: 0px;
  text-transform: none;
  cursor: text;
}

.c1 {
  display: flex;
  position: absolute;
  justify-content: center;
  align-items: center;
  width: 100%;
  font-weight: 400;
  letter-spacing: 0.05em;
  transition: 0.5s;
  opacity: 1;
  transform: translateY(48px);
}

<div>
  <p
    class="c0 c1"
    color="#23242a"
  />
</div>
`;
